dockerfiles/theia/src/start.sh

```diff
@@ -10,6 +10,19 @@
 
 if [ -z "$THEIA_PORT" ]; then
     export THEIA_PORT=3000
+else
+    # Parse THEIA_PORT env var in case it has weird value, such as tcp://10.108.137.206:3000
+    theia_port_number_regexp='^[0-9]+$'
+    if ! [[ "$THEIA_PORT" =~ $theia_port_number_regexp ]]; then
+        # THEIA_PORT contains something other than a number
+        theia_port_uri_regexp='^tcp://[0-9a-zA-Z.]+:[0-9]+$'
```

Comment:
I would name it like `theia_port_tcp_uri_regexp` because of hardcoded value.

Reply:
done